presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -177,6 +178,10 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
                             domainTranslator.toPredicate(newDomain.simplify().transform(assignments::get))));
             constraint = new Constraint(newDomain, evaluator::isCandidate);
         }
+        else if (newDomain.isAll() && !usesLegacyTableLayouts) {
+            // for legacy layouts the only way of passing desired columns during planning is via ConnectorMetadata.getTableLayouts
+            return Optional.empty();
```

Comment:
The comment sounds as if you'd like to return here for legacy layouts, but you're returning for non-legacy layouts. I don't understand.

What's the point of this additional condition in the `if`?
If domain is all and there is no evaluator involved, what's the point in continuing?

Reply:
discussed offline